src/main/java/co/rsk/altbn128/cloudflare/Utils.java

```diff
@@ -11,7 +12,11 @@ public static boolean isMac() {
         return (OS.indexOf("mac") >= 0);
     }
 
-    public static boolean isLinux() {
-        return (OS.indexOf("nux") >= 0);
+    public static boolean isLinuxAmd64() {
```

Comment:
Im wondering that it could be a compatible cpu, but `OS_ARCH` might now include the `Amd` part..

also for ARM cpu I believe we could have `aarch64` as a valid value.

so do you think we could check if it's an arm based cpu first, and if not then fall back to default amd?.. Eg. something like this?..

```java
if (isLinux()) {
  if (OS_ARCH.contains("arm") || OS_ARCH.contains("aarch")) {
    // use ARM ver.
  } else {
    // use default AMD ver. ( current one )
  }
}
```

Reply:
could you please add some unit tests?

Reply:
Done.

There is not much we can do with testing unless we use static mocks. This is tightly couple to the OS.